contentcuration/contentcuration/views/settings.py

```diff
@@ -34,22 +38,69 @@
 from contentcuration.forms import PreferencesSettingsForm
 from contentcuration.forms import ProfileSettingsForm
 from contentcuration.forms import StorageRequestForm
+from contentcuration.serializers import UserChannelListSerializer
 from contentcuration.models import Channel
+from contentcuration.models import User
 from contentcuration.tasks import generateusercsv_task
 from contentcuration.utils.csv_writer import generate_user_csv_filename
 from contentcuration.utils.google_drive import add_row_to_sheet
 from contentcuration.utils.policies import get_latest_policies
+from contentcuration.utils.messages import get_messages
 
 ISSUE_UPDATE_DATE = datetime(2018, 10, 29)
 
+PREFERENCES = "user_preferences"
+CURRENT_USER = "current_user"
+MESSAGES = "i18n_messages"
 
 @login_required
 @browser_is_supported
 @has_accepted_policies
 def settings(request):
     if not request.user.is_authenticated():
         return redirect('accounts/login')
-    return redirect('settings/profile')
+    else:
+        anon = request.user.is_anonymous()
+        current_user = (
+            None
+            if anon
+            else json_for_parse_from_serializer(UserChannelListSerializer(request.user))
+        )
+        preferences = DEFAULT_USER_PREFERENCES if anon else request.user.content_defaults
+    return render(
+        request,
+        'settings.html',
+        {
+            CURRENT_USER: current_user,
+            PREFERENCES: json_for_parse_from_data(preferences),
+            MESSAGES: json_for_parse_from_data(get_messages()),
+        },
+    )
+
+@login_required
+@api_view(['PATCH'])
```

Comment:
Not a big deal, but is there a reason not to use POST here? Most of the other endpoints like this are POST only, so it might be good to standardize as much as possible

Reply:
I used PATCH because we're modifying an existing resource whereas POST is typically for creating a new resource.